### PR TITLE
Ft implement sign up functionality

### PR DESCRIPTION
**Pull request by MutegekiHenry**
## Description

- Refactor Register Page and removed unused phone field

- Added bootstrap and implemented it on login/signUp forms

- Fix broken links on signUp and Login Page

- Add Axios and use it to make calls to API for the login and register routes

- Handle success and failure on login and register pages 

PS: The current implementation of the Axios calls require the API to change to `http`

Fixes ( #36) and (#35)

## How Has This Been Tested?
This branch has been run on localhost in Linux OS and Mozilla browser and Chromium and works as expected

## Screenshots (if applicable, else remove this line / section)

Register Page Before this PR
Screenshot from 2020-05-28 22-43-31

Register Page after this PR with the ability to talk to back-end API
Screenshot from 2020-05-28 22-43-47

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have added necessary inline code documentation
- [ ] I have added tests that prove my fix is effective and that this feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by pearlonzo — APPROVED** [review]

**Review by pearlonzo — DISMISSED** [review]

**Review by pearlonzo — APPROVED** [review]

**Review by pearlonzo — APPROVED** [review]

File: src/components/LoginPage/index.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { Link } from 'react-router-dom';
 import './LoginPage.css';
 import Header from '../PageHeader';
 import LandingFooter from '../PageFooter';
+import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';
```
Comment (geneowak) [inline src/components/LoginPage/index.js:7]: Let's use material-ui

File: src/components/LoginPage/index.js [hunk 2]
```diff
@@ -32,11 +34,46 @@ class Login extends Component {
...
         email: '',
         password: ''
-      }
+      },
+      error: ''
+
     };
   }
+
+  componentDidMount() {
```
Comment (geneowak) [inline src/components/LoginPage/index.js:43]: Let's use functional components

**Review by onabright — APPROVED** [review]
This looks good.

File: src/components/signup/SignUp.js [hunk 3]
```diff
@@ -4,6 +4,8 @@ import { Link } from 'react-router-dom';
 import './SignUp.css';
 import Header from '../PageHeader';
 import LandingFooter from '../PageFooter';
+import axios from 'axios';
+import '../../../node_modules/bootstrap/dist/css/bootstrap.min.css';
```
Comment (geneowak) [inline src/components/signup/SignUp.js:8]: See if material ui would make it easier...

File: src/components/signup/SignUp.js [hunk 4]
```diff
@@ -73,97 +98,80 @@ class SignUp extends Component {
...
+            </div>
+            <div className="form-group">
+              <label htmlFor="fullname">Full Name</label>
+              <input
+                className={formErrors.fullName.length > 0 ? 'error' : null}
+                type="text"
+                name="fullname"
+                placeholder="Full Name"
+                required
+                onChange={this.handleChange}
```
Comment (geneowak) [inline src/components/signup/SignUp.js:120]: Try material ui components
https://material-ui.com/components/text-fields/#

File: src/components/signup/SignUp.js [hunk 5]
```diff
@@ -73,97 +98,80 @@ class SignUp extends Component {
...
+                placeholder="Confirm password"
+                required
+                formNoValidate
+                onChange={this.handleChange}
+              />
+              {formErrors.password.length !== formErrors.confirmPassword.length && (
+                <span className="errorMessage">{formErrors.password}</span>
+              )}
+            </div>
+            {error && <span className="errorMessage">{error}</span>}
```
Comment (geneowak) [inline src/components/signup/SignUp.js:165]: you wouldn't need this with material ui

**Review by geneowak — CHANGES_REQUESTED** [review]

**Pull request merged**